### ISSUE TITLE
Confused about training!

### ISSUE THREAD

**Issue by AhmadM-DL** [facebookresearch/deepcluster#45]
Hello there,

The Alexnet model you defined creates a network with the final layer having 1000 classes by default. And this can be seen in the following piece of code in models/alexnet.py:95:

```
def alexnet(sobel=False, bn=True, out=1000):
    dim = 2 + int(not sobel)
    model = AlexNet(make_layers_features(CFG['2012'], dim, bn=bn), out, sobel)
    return model
```

On the other side the default number of clusters is set to 10,000 in main.py:38:

```
parser.add_argument('--nmb_cluster', '--k', type=int, default=10000,
                    help='number of cluster for k-means (default: 10000)')
```

**How it is possible to train a network with 1000 classes using pseudo labels of 10,000 clusters?**

**Comment by AhmadM-DL** [comment]
Sorry,
I just found that you recreate the last layer using the following code piece in main.py:170

 ```
   # set last fully connected layer
        mlp = list(model.classifier.children())
        mlp.append(nn.ReLU(inplace=True).cuda())
        model.classifier = nn.Sequential(*mlp)
        model.top_layer = nn.Linear(fd, len(deepcluster.images_lists))
        model.top_layer.weight.data.normal_(0, 0.01)
        model.top_layer.bias.data.zero_()
        model.top_layer.cuda()
```
